File: applications/app-bemyplan/src/main/java/com/deploy/bemyplan/auth/controller/dto/response/LoginResponse.java
package com.deploy.bemyplan.auth.controller.dto.response;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

@ToString
@Getter
@NoArgsConstructor(access = AccessLevel.PRIVATE)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class LoginResponse {
    private String token;

    private String sessionId;

    private Long userId;

    private String nickname;

    public static LoginResponse of(String token, String sessionId, Long userId, String nickname) {
        return new LoginResponse(token, sessionId, userId, nickname);
    }
}

File: applications/app-bemyplan/src/main/kotlin/com/bemyplan/auth/LoginResponse.kt
package com.bemyplan.auth

data class LoginResponse(
    val token: String,
    val sessionId: String,
    val userId: Long,
    val nickname: String,
) {
    companion object {
        fun of(token: String, sessionId: String, userId: Long, nickname: String): LoginResponse {
            return LoginResponse(token, sessionId, userId, nickname)
        }
    }
}